src/workerd/jsg/jsvalue.h

```diff
@@ -255,6 +267,20 @@ class JsDate final : public JsBase<v8::Date, JsDate> {
   using JsBase<v8::Date, JsDate>::JsBase;
 };
 
+class JsPromise final : public JsBase<v8::Promise, JsPromise> {
```

Comment:
These can probably be handled just like the `Symbol`, `BigInt`, and other similar simple cases. Look below for the `V(Symbol)`, `V(BigInt)`, etc.

I am a bit concerned that someone could confuse `jsg::Promise` with `jsg::JsPromise` but that's minor. Perhaps a comment in here distinguishing the two would be helpful tho?

Reply:
ah, right, ok, yeah I missed that. Good as is then. I would definitely suggest adding some comments to highlight to folks that `jsg::JsPromise` is definitely not the same thing as `jsg::Promise`.